Mage/src/main/java/mage/constants/AbilityType.java

```diff
@@ -1,26 +1,31 @@
 package mage.constants;
 
 /**
- * @author North
+ * @author North, Susucr
  */
 public enum AbilityType {
-    PLAY_LAND("Play land", true),
-    MANA("Mana", false),
-    SPELL("Spell", true),
-    ACTIVATED("Activated", false),
-    STATIC("Static", false),
-    TRIGGERED("Triggered", false),
-    EVASION("Evasion", false),
-    LOYALTY("Loyalty", false),
-    SPECIAL_ACTION("Special Action", false),
-    SPECIAL_MANA_PAYMENT("Special Mana Payment", false); // No activated ability and no special action. (e.g. Improvise, Delve)
+    PLAY_LAND("Play land", true, false, false),
+    SPELL("Spell", true, false, false),
+    STATIC("Static", false, false, false),
+    EVASION("Evasion", false, false, false),
+    ACTIVATED_NONMANA_NONLOYALTY("Activated", false, true, false),
+    ACTIVATED_MANA("Mana", false, true, true),
+    ACTIVATED_LOYALTY("Loyalty", false, true, false),
+    TRIGGERED("Triggered", false, false, false),
+    TRIGGERED_MANA("Triggered Mana", false, false, true),
```

Comment:
Side note: we may need a similar cleanup for `TRIGGERED` + `TRIGGERED_MANA`, now that MANA has been split in two.